profiler/src/ProfilerEngine/Datadog.Profiler.Native/AllocationsProvider.cpp

```diff
@@ -21,12 +21,15 @@ AllocationsProvider::AllocationsProvider(
     IFrameStore* pFrameStore,
     IThreadsCpuManager* pThreadsCpuManager,
     IAppDomainStore* pAppDomainStore,
-    IRuntimeIdStore* pRuntimeIdStore)
+    IRuntimeIdStore* pRuntimeIdStore,
+    IConfiguration* pConfiguration)
     :
     CollectorBase<RawAllocationSample>("AllocationsProvider", pThreadsCpuManager, pFrameStore, pAppDomainStore, pRuntimeIdStore),
     _pCorProfilerInfo(pCorProfilerInfo),
     _pManagedThreadList(pManagedThreadList),
-    _pFrameStore(pFrameStore)
+    _pFrameStore(pFrameStore),
+    _sampleLimit(pConfiguration->AllocationSampleLimit()),
+    _sampler(pConfiguration->AllocationSampleLimit(), pConfiguration->GetUploadInterval().count() * 1000)
```

Comment:
maybe pass the upload interval and leave the conversion in the ctor (or later)

Reply:
✅ done